frontend_app_gallery/src/component/MainPreview.jsx

```diff
@@ -2,17 +2,24 @@ import React from 'react'
 import PropTypes from 'prop-types'
 
 class MainPreview extends React.Component {
+  constructor (props) {
```

Comment:
a constructor that does nothing should be remove it.
Although, bellow, you setState height and width so give default values in the constructor